node/api/apistruct/struct.go

```diff
@@ -0,0 +1,26 @@
+package apistruct
+
+import (
+	"context"
+
+	lotusstruct "github.com/filecoin-project/lotus/api/apistruct"
+	logging "github.com/ipfs/go-log/v2"
+
+	"github.com/filecoin-project/sentinel-visor/node/api"
+)
+
+var log = logging.Logger("lily-api")
+
+type LilyNodeStruct struct {
+	lotusstruct.FullNodeStruct
```

Comment:
We don't need all of this. What is the minimum we need?

Reply:
We need the following API's from lotus:
- `common.CommonAPI` for the RPC and authentication/permissions.
- `full.ChainAPI` for chain event notification and chain inspection.
- `full.StateAPI` for actor state extraction and inspection.

I have redeclared the `LilyAPI` to require only these.